src/Dapr.Client/StateQueryResponse.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System.Collections.Generic;
+
+namespace Dapr.Client
+{
+    /// <summary>
+    /// Represents the response from a state query.
+    /// </summary>
+    public class StateQueryResponse<TValue>
+    {
+        /// <summary>
+        /// The results of the query.
+        /// </summary>
+        public IReadOnlyList<StateQueryItem<TValue>> Results { get; set; }
+
+        /// <summary>
+        /// The pagination token to continue the query.
+        /// </summary>
+        public string Token { get; set; }
+
+        /// <summary>
+        /// The metadata to be passed back to the caller.
+        /// </summary>
+        public IReadOnlyDictionary<string, string> Metadata { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an individual item from the results of a state query.
+    /// </summary>
+    public class StateQueryItem<TValue>
+    {
+        /// <summary>
+        /// The key from the matched query.
+        /// </summary>
+        public string Key { get; set; }
+
+        /// <summary>
+        /// The data of the the key from the matched query.
+        /// </summary>
+        public TValue Data { get; set; }
+
+        /// <summary>
+        /// The ETag for the key from the matched query.
+        /// </summary>
+        public string ETag { get; set; }
+
+        /// <summary>
+        /// The error from the query, if one occurred.
+        /// </summary>
+        public string Error { get; set; }
```

Comment:
What do users do with this?

Reply:
This seems like a bonkers API design no? Would selected items ever fail to decrypt or would everything fail uniformly.

Reply:
As we discussed, everything decrypts individually,  hence the individual errors.